### DISCUSSION TITLE
2 PC Stream issue: Black screen on streaming PC: VERY LONG and DETAILED POST

### DISCUSSION THREAD

**Discussion by EmmanuelReyTan** [DistroAV/DistroAV#643] [General]
I checked all the issue both open and closed to see if it is already fix or already posted:
#31 and #32 solution didn't work for me (more info below).
#49 and #50 kinda same problem but I am using windows.
#54 kinda same problem but with different setup (more info below).
#58 not the same problem because my PC1(gaming pc) **SHOWS UP** on the sources on the PC2(streaming pc) but showing a black screen.

I have 1 Gaming PC and 1 Laptop for streaming.

**Setup**
Both my pc and laptop are connected in the same network that support gigabit connection.
Both are using Lan Cable Cat5e that support gigabit connection
Both are running in windows 10
Both are running OBS 20.0.1 64 bit version
Both are running obs-ndi-4.1.3
Both have a resolution of 1364x768
Both windows 10 are up to date including drivers.
Both are using intel core I5 and base on my research it support ssse3

**Specs**
Gaming PC
CPU: Intel I5 6th Gen
GPU: GTX 750ti
Ram: 8 GB
HDD: 1 T

Streaming Laptop
CPU: Intel I5-3210M @2.50GHz
GPU: Nvidia GeForce GT 630m
Ram: 4 GB
HDD: 500 GB

**Problem**
NDI Source on my **STREAMING PC** can discover my **GAMING PC** but it only shows a black screen with a red dot on top left corner.

NDI Source on my **GAMING PC** can discover my **STREAMING PC** and it works properly with no problem.

**I tried:**
1. To disable windows firewall on both pc like the solution on #31 and #32 , but the problem still exist.
2. To used the NDI studio monitor on my streaming pc and it works perfectly fine (it can capture both video and audio), but the obs still showing a black screen.
3. To connect the lan cable directly from my pc to my laptop but the problem still exist (so its not a router issue).
4. To check low bandwidth, problem is still there.
5. Re-install both the obs and obs-ndi, but problem is still there.
6. Disable avast anti virus on my gaming pc (because my avast has a firewall setting so I thought it might be the problem), but still the problem didn't go away. Streaming PC is freshly installed so there are no antivirus and have plenty of free space.
7. Change my router (my ISP company provided me a new router LOL). problem is still there.
8. Tried to format and Upgrade OS on my streaming PC, when i first tried the obs-ndi-4.1.3 the other day, my streaming PC has a windows 7 ultimate and it shows black screen so I thought it might be because I am running a windows 7 version when my gaming PC is running windows 10. So I decided to format my HDD and install a fresh windows 10, but with no luck the problem still exist.
9. Tried to used a static IP and change my dns (because one of the users from 1 of the post suggested it) but problem is still there.
10. used a new cat5e lan cable (included from the new router provided but my ISP), problem is still there.

IMPORTANT NOTES:
there is a 1 time, for some reason my streaming pc suddenly shows a video instead of a black screen (shows my gaming PC Display) but after a couple of minutes the video freezes, when I tried to delete and add NDI source, black screen shows up again.

feel free to ask more about it or if I can provide more information like logs or something.

**Comment by EmmanuelReyTan** [comment]
If I for some reason break a rules of any sort feel free to delete my post, no hard feelings :)

**Comment by Palakis** [comment]
> If I for some reason break a rules of any sort feel free to delete my post, no hard feelings

Being comprehensive doesn't break the rules! I'm glad there are posts like this.

**Comment by beakergoesmeepmeep** [comment]
Seemingly the similar issue for me. Tried the same things here and linked with the same results, save one extra observation below.

Sources tried:
OS X with Newtek NDI Scan Converter - Various windows and full screen.
OS X with Vmix DekstopCapture - Various windows and full screen.
Arch Linux OBS NDI Output
iPhone 6s with Newtek NDI Camera App

In all cases Newtek NDI Monitor running on the OS X machine would display the sources fine in either low or high bandwidth modes. But latest released OBS and OBS-NDI on both Arch Linux and OS X machines would not display anything.

I also went over my network router setup in detail. Found and fixed some minor issues with mDNS. No difference.

At one time in several hours of trying things it did start working apparently randomly on the Arch Linux machine. When it did work it seemed solid and I could connect and disconnect sources, change settings etc. After then shutting down and restarting OBS Studio I have not been able to get any video since.

**Comment by wrightyrx7** [comment]
Make sure you have a source capturing the desktop or a game on OBS on the GAMING PC when you add the NDI source on the STREAMING PC.

Im not sure if this is the exact reason but i have definitely had the same problem and think the above is how i sorted it. 

Im not 100% the above was the reason because it was when i was first setting everything up and i havnt really touched anything since. But i do know it did it to me a few times and i always managed to sort it. 

If it happens again and the solution above isnt correct i will be sure to come back and post.

**Comment by wazerstar** [comment]
Right click the red dot > transform > fit to screen, no more black screen.

I bet its because your capture screen is minimized aka moved all along X/Y to the top left corner.

**Comment by beakergoesmeepmeep** [comment]
Already tried that but just checked again to be sure. No dice.

**Comment by wazerstar** [comment]
mark the red dot hit ctrl+e take print screen and show here.

**Comment by wazerstar** [comment]
So status?, I guess its always the same with people not answering back on own error40 human errors.

**Comment by beakergoesmeepmeep** [comment]
When I tried that last it crashed OBS. Not had time to do anything more since. I plan to have a look with Wireshark what is going on with the connection. Just too busy right now.

**Comment by EmmanuelReyTan** [comment]
Sorry about that. I can't really provide useful information/feedback right now because I am working out of town therefore I can't access my PC but I will be back this weekend to try those step and hope it will fix it, thank you and have a nice day.

**Comment by EmmanuelReyTan** [comment]
I would like to update my post, the problem is still there, sadly to say the solution gave by wazerstar didn.t work on me, But I kinda have a work around on my problem. 

Since I'm doing a 2 pc stream, the thing I am doing right now is using the in-home streaming of steam on PC2, basically I play on PC1 while using the in-home streaming of steam on PC2 while running the obs at the same time on PC2, so basically it captures the in-home streaming of my PC2. Problem is I can't used PC2 to check for my twitch chat or others.. I will just do this for the time being until Mr. Palakis finally found a solution for this problem.

**Comment by cs2383** [comment]
Had the same Issue, disabled all network adapters, left only the one connecting to the router. Problem solved.

**Comment by TheWookieWay** [comment]
I'm pretty sure this is an OBS issue, not an OBS-ndi specific problem, that I get occasionally, but always seem to resolve, but never remember how.
It may be necessary to have a source streaming when you setup NDI in the first place on streaming pc.  removing and re-adding the source seems to fix it most of the time, but I do remember times when just that alone does not help.  I will pay attention and research the bug when i come across it again.

**Comment by borgeesalisson** [comment]
First thanks for the explanation, but I'm encountering problems, I've already followed the tutorial and reinstalled everything from scratch. I'm using a gaming PC and a notebook for streaming, both with windows 10 64 bit, but all I get is a black screen with a red dot in the left corner, even resizing to full screen it remains black. I added obs with firewall exception, my connection shows in status as being gigabit, I checked the network adapters and checked that there are several that were disabled leaving only the main active, but all I try to do is get a black screen, I'm finding that the problem may be my windows that is not original on both machines, could this be the black screen motif? Thank you in advance for your attention.

**Comment by nimhrod** [comment]
I had the same issue and JUST resolved it taking these few steps.

1. I recreated a homegroup for both computers to join.
2. I disabled all other network adapters besides the ethernet ports (re-enabled them afterwards and the problem stayed away).
3. Highlighted the NDI source on streaming pc and pressed "ctrl+e". I clicked the reset button on the bottom which changed the size to 1920x1080

It appeared immediately after that!

Now I just have to get rid of the lag and stuttering. Probably due to the lack of a cat5e cable.

**Comment by harvestcore** [comment]
I have the same issue, but in my case my main pc is running Windows and my streaming pc is running Ubuntu. I have been trying everything you said but the problem is still there.

**Comment by beakergoesmeepmeep** [comment]
I'd originally given up on this but I just tried this technology preview from Orfast http://ndi.orfast.com/download.html and it worked straight out of the box no issues on video or audio in high or low bandwidth. On the same machine I was trying OBS on. Now I've got a good and bad to compare on the same platform I'll have another look. Would be good to know if this other product works for others.

**Comment by EmmanuelReyTan** [comment]
Hi everyone, if you are still encountering the problem, I have a quick fix for it. You just need to download the NDI tools and use the NDI Monitor on your streaming pc then change the source of it to your gaming pc. This will then preview the display of your gaming pc. Now on OBS make a display capture or window capture. and capture the NDI Monitor.

Downside is
I still can't figure how to change the resolution, in my case it is low quality around 720p.

**Comment by jbrouhard98** [comment]
This problem is persistent on Arch Linux with arch being the stream rig, windows being the source.  I can pull from Arch to Windows with ndi source no problem, but if i'm pulling ANY NDI source onto the linux box, i bget nothing.   It's been extremely hit and miss, and OBS logs show this

10:11:43 AM.380: [obs-ndi] can't create a receiver for NDI source ''

10:11:43 AM.384: User added source 'NDI™ Source' (ndi_source) to scene 'Games'

10:11:50 AM.431: [obs-ndi] started A/V threads for source 'DESKTOP-DSETQMB (WindowsRig)'

10:11:50 AM.431: [obs-ndi] started A/V threads for source 'DESKTOP-DSETQMB (WindowsRig)'

10:11:50 AM.431: [obs-ndi] audio thread for 'NDI™ Source' started

10:11:50 AM.431: [obs-ndi] started A/V threads for source 'DESKTOP-DSETQMB (WindowsRig)'

10:11:50 AM.431: [obs-ndi] audio thread for 'NDI™ Source' started

10:11:50 AM.431: [obs-ndi] video thread for 'NDI™ Source' started

**Comment by milanthkr** [comment]
I was facing similar issues and disabling IPv6 on the network adapter of my gaming PC worked for some reason.

**Comment by jbrouhard98** [comment]
That worked, to some extent.  I have to constantly destroy the source, and recreate it every time obs is loaded.  Not sure if this is a bug with NDI or with OBS itself.

Thanks for the suggestion milanthkr !!

**Comment by HeliumStudios** [comment]
Hey ! I found a solution by tweaking my network properties.
On your gaming PC, go on network properties, set your network profile to Private and deactive limited connection (Windows 10).
Hope it helps you :D

**Comment by pedraodeus** [comment]
hi there, problem:
gaming pc with ndi output enable,
streaming notebook, cant find my gaming pc in NDI Source. (source name in blank)
cant discover my gaming/streaming pc ={


both with:
ndi, obs last version.
firewall off/on
network cards are gigabit capable
router wired CAT5E
windows 10 brand new installed.
ip automatic
same gateway..

already tested everything... dont know what to do anymore.. =//

anyone have suggetions?
pls

**Comment by pedraodeus** [comment]
The problem was the modem!
With a switch works fine!

**Comment by Palakis** [comment]
@peepersoak does this issue still happens?

**Comment by zurgalos** [comment]
@Palakis I still have this problem. I literally tried everything.

**Comment by justweb1** [comment]
I had the same problem. All I did to fix it is to make sure that my network connection was set to 'Private Network'

**Comment by shairez** [comment]
[SOLVED]
Yep. changing from `Private` to `Public` in Windows 10's network settings did the job!

**Comment by VeryG00dName** [comment]
I had the same problem tried everything said before half of it I had already done an it still won't work.

**Comment by TeraBrite** [comment]
I'm having pretty much the same issue.

PC 1 (NDI Output): 
Alienware Laptop
OS: Windows 10 Pro 64 bit
CPU: Intel Core i9-8950HK
RAM: 32 GB
GPU: RTX 2080 With Max-Q
Connection: Killer E2500 Gigabit Ethernet Controller
OBS Output: 1920 x 1080 59.94 fps (Scaled from 4k 59.94 base canvas)

PC 2 (Input with NDI Source)
DigitalStorm Desktop
OS: Windows 10 Pro 64 bit
CPU: i9 7940X
RAM: 64 GB
GPU: RTX Titan
Connection: ASUS Prime X299 Deluxe Onboard Gigabit LAN
OBS Output: 1920 x 1080 59.94 fps (Scaled from 4k 59.94 base canvas)

I'm using the latest version of OBS and the latest version of the NDI plugin on both systems. I'm able to get my PC 1 to show up as a source on PC 2 but when I choose it, like everyone else said it's just a red dot and resizing to fit the screen does nothing other than make it a transparent red rectangle with no content. I'm pretty sure I've tried all of the suggestions on this page including one I found on a video where he walks you through enabling streaming PCs in settings for Windows 10. For some reason I've noticed I'm only able to get it to show up as a source if I turn OBS on PC 2 on first then turn OBS on PC 1 on with the NDI output already enabled. Usually if I turn PC 1 on first it doesn't show up for some reason.

Even though I have no need to, I noticed that when I run PC 2 as an output, I'm not able to see it as a source on PC 1 at all.. however I am able to see PC 2 as a source on itself. Kinda useless but it works. I'm not able to do see PC 1 as a source on itself though. Also I tested using my older and much slower Alienware laptop as the NDI output and it works flawlessly, so clearly there's something not quite right on my new Alienware laptop causing it to not work properly with NDI.

Some things I tried on the laptop:
1. Uninstalled/reinstalled OBS and NDI Plugin
2. Disabled IPv6
3. Enabled Streaming in Windows 10
4. Disabled all network adapters other than the LAN port
5. Allowed OBS on Windows Defender Firewall and Malwarebytes

Any help would be greatly appreciated!

**Comment by rigofdz** [comment]
Just adding my .2 cents after spending a few hours getting this to work.  Maybe it will save a few people some hours of troubleshooting .   

My systems: Thinkpad T420 running Windows 7 and Surface 4 Pro on Windows 10.  After trying all the suggestions above what finally got this to work was the following actions on my Windows 7 box(output source).  Not sure which did the trick but here they are:  
-Locate OBS Studio exe file and configure it to run it in Compatibility mode for Windows 7 and Select the Checkmark to run it as administrator.   
-I uninstalled Comodo Firewall 
-Restarted

Happy to say this is now working :).

**Comment by BlueBoxBlueSkyLemonJuice** [comment]
So I tried some if not most of these things to try to get NDI to work. It started working again when I set both computers to private, made sure they were on the same network (wifi/ethernet) then disabled the firewall for private on both.

**Comment by marcusademola** [comment]
Hi,
I am also witnessing that problem, where **the NDI camera is NOT shown on OBS 25.0.4** (64bit Win10). See screen1.

Here (screen1) how the problem manifests on OBS 
![screen1-NDIsourceNotShowingOBS-problem](https://user-images.githubusercontent.com/18284093/79367069-83488a00-7f4d-11ea-9f36-d0e5d3d43497.PNG)

Here some workaround (screen2) after selecting "NDI video out" OBS is able to "recaputure" the stream from local network.  
![screen2-NDIsourceNotShowingOBS-butnowitworks](https://user-images.githubusercontent.com/18284093/79367356-0538b300-7f4e-11ea-97b9-b6d881f58219.PNG)


But this is not a solution from my point of view, even Video shows well on the NDI Analysis (and Studio Monitor). Except with some latency on Wifi. 

I also removed all firewall blockers from the OBS, so its not problem of the network. 

Any help would be welcomed , about setting that enable to direct presentation of video from NDI camera on OBS (without workarounds, like screen2)

**Comment by hybridherbst** [comment]
What worked for me was disabling all but the actually used ethernet adapter - there might be not just the "phyiscal" adapters but virtual ones as well (from Docker, VMWare, ...) interfering with NDI streams. Looks like that at least is something that could be solved on sender/receiver end - NDI's own tools don't suffer from this.

**Comment by PocketNinjaDesign** [comment]
I'm going to show something that fixed it for me.

My setup was working for ages, then I didn't use it for a while and somehow maybe with Windows updates the firewall settings had messed up.

I'm going to give credit to this guy
https://www.youtube.com/watch?v=0uzfSpX1ubE

But basically on the output computer I had to set OBS for Firewall to be checked true for both Public and Private.  The same is set on my Streaming computer as well.  After reading through this whole thread and looking elsewhere the Firewall was the issue for me!  I've reset my Windows Compatibility mode to off and run as admin.  Just felt dirty still having that, it all still works.

Bosh!!!

**Comment by afiaka87** [comment]
Ugh, this was stupid of me but I found a solution.

The Elgato website specifically states:

> 4)  In OBS Studio, click the “+” button in your scene and add a new “NDI source”.
> 5)  Give the source a name like “4K Capture Utility Stream Link” then click OK.

This is of course incorrect. What you should actually do is click the dropdown and select the existing source.

**Comment by mavkeysound** [comment]
Hey Guys i do this solution (and another solutions) but nothing works. I want to send video from notebook to pc but only detect source but i have blank screen. Finally i check the ports with CMD with the command netstat -a and discover that the ndi ports is redirecting to address 192.168.137.1 (sometimes i share internet with that ethernet port but i dont know why if i deactivate the internet sharing internally is not deactivate.... well i must to reactivate and desactivate to finally dont apear 192.168.137.1 in status of ethernet (two ipv4 there). Then i check with netstat -a and finally the ports redirecting to correct adress and works!!!! Finally!!!!. (For it is help for something... now if i reboot my notebook still have the 192.168.137.1 adress and i must to disable the service... i dont know why.... sorry my bad english)

**Comment by becmer** [comment]
In my case I got black screen because of two Hyper-V Virtual Ethernet Adapters being enabled. Disabling them solves problem. The core problem has been described here: https://stackoverflow.com/q/39749092/854291

**Comment by HeyImSylo** [comment]
Hey guys,

I tried pretty much everything now to try to fix it, but I stilll get the black screen on the streaming pc, even after I click "fit to screen". The source from the gaming pc shows on the streaming pc obs but I only see the black screen.
I have Windows 10 on both PCs and good enough specs to be using OBS + NDI. I also use the same monitor resolution on both PCs. I don't know what to do anymore.

I tried:
- reinstall and reset OBS and NDI-Plugin on both PCs
- change the network profile to private on both PCs
- enable media streaming on both PCs
- enable both private and public settings on the firewall on both PCs
- disable all network adapters expect the one LAN connection to the router on each PC
- reboot router

**Comment by Lullyfito** [comment]
Hey guys , i am with black screen problem as well , i did a lot of things , like calling my network provider to change my ip to fix  , open those ports that ndi required like 5353 and 5960 , firewall and etc. So after that when i was giving up i disabled all my networks conections and enable just the one that i am using in the moment , and after 1 sec my screen shown up on studio monitor and stream labs as well .(sorry for my english , hope that helps anyone)

**Comment by bigfroge** [comment]
Hey guys I am having basically the same issue most people seem to be having in this discussion. I feel as if I have tried most solutions people have given with some results but nothing that fixed the problem. I am currently looking at a black screen in OBS/streamlabs but the NDI source is picking up audio just no video. I am willing to post logs or whatever is need to solve this problem. Help would be appreciated

**Comment by KINGMARSTX** [comment]
Hello everyone, I believe I have found a resolution to this issue. (Issue lies in having integrated gpu and cpu graphics)

I tried EVERYTHING above, nothing worked. I was getting audio, but just a black screen.

I found here https://forums.newtek.com/threads/no-video-from-ndi-scan-converter-on-tc8000.162703/ that by forcing Newtek to my integrated graphics using Nvidia Control Panel, I was able to pull visual via NDI!

Hope this helps!
